### DISCUSSION TITLE
How to normalized the grayscale of a image texture

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#464] [Q&A]
![image](https://user-images.githubusercontent.com/36077492/209838685-81a961d5-19e3-4a24-88ae-23d9abebfa06.png)

For example, nodes above. How to normalize the texture grayscale to :  from 0 to 1

**Comment by pragma37** [comment]
Sorry, I don't think I understand the question.

If the image color is in the 0 to 1 range, the grayscale should be too.

**Comment by HSDSZ** [comment]
For example
![image](https://user-images.githubusercontent.com/36077492/209962249-524f06ca-7001-4de9-8682-f938bb519b3f.png)

The gray scale nodes return 0 at the black area. A none 0 value at the gray area.

I want to somehow normalize the grayscale over the whole image. and want it be 0 at black area and 1 at the gray area. Is this doable?

**Reply by Kolupsy** [reply]
For that you would need to find the darkest and lightest pixel of that image. That can be pretty expensive and unnecessary to do constantly especially since it might be a ton of pixels to go through and also the pixel values of images usually never change. 

So better try to precompute this remapping or do it with lo precision by for example sampling a couple random pixels in the texture or a few in a grid

**Reply by HSDSZ** [reply]
> For that you would need to find the darkest and lightest pixel of that image. That can be pretty expensive and unnecessary to do constantly especially since it might be a ton of pixels to go through and also the pixel values of images usually never change.
> 
> So better try to precompute this remapping or do it with lo precision by for example sampling a couple random pixels in the texture or a few in a grid

Ok, I see. thanks
